src/category_theory/limits/shapes/pullbacks.lean

```diff
@@ -0,0 +1,244 @@
+-- Copyright (c) 2018 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+import category_theory.eq_to_hom
+import category_theory.limits.cones
+import category_theory.sparse
+
+open category_theory
+
+namespace tactic
```

Comment:
Is this actually being used by `tidy` in the end? I do see lot's of manual `cases` left in the file.
Does this merit putting it in its own file, and only putting the `local attribute` line in this file? I can imagine that it is also useful elsewhere.

Reply:
It is actually being used, in quite a few places. I've moved this definition to `src/tactic/basic.lean`.